packages/react-components/react-field/src/components/Field/Field.test.tsx

```diff
@@ -0,0 +1,126 @@
+    const MockFieldLabelledBy: ForwardRefComponent<MockFieldProps> = React.forwardRef((props, ref) => {
+      const state = useField_unstable(props, ref, {
+        component: MockComponent,
+        classNames: mockFieldClassNames,
+        labelConnection: 'aria-labelledby',
+      });
+      useFieldStyles_unstable(state);
+      return renderField_unstable(state);
+    });
+
+    const result = render(<MockFieldLabelledBy label="Test label" />);
+    const input = result.getByRole('textbox');
+    const label = result.getByText('Test label') as HTMLLabelElement;
+
+    expect(label.id).toBeTruthy();
+    expect(input.getAttribute('aria-labelledby')).toBe(label.id);
+    expect(label.htmlFor).toBeFalsy();
+  });
+
+  it('adds a required asterisk * to the label when requried is set', () => {
+    const result = render(<MockField label="Test label" required />);
+
+    expect(result.getByText('*')).toBeTruthy();
+  });
+
+  it('sets aria-describedby to the hint', () => {
+    const result = render(<MockField hint="Test hint" />);
+    const input = result.getByRole('textbox');
+    const hint = result.getByText('Test hint');
+
+    expect(hint.id).toBeTruthy();
+    expect(input.getAttribute('aria-describedby')).toBe(hint.id);
+  });
+
+  it('sets aria-describedby to the validationMessage if not an error', () => {
+    const result = render(<MockField validationMessage="Test validation message" validationState="warning" />);
+    const input = result.getByRole('textbox');
+    const validationMessage = result.getByText('Test validation message');
+
+    expect(validationMessage.id).toBeTruthy();
+    expect(input.getAttribute('aria-describedby')).toBe(validationMessage.id);
+  });
+
+  it('sets aria-describedby to the hint + validationMessage if not an error', () => {
+    const result = render(
+      <MockField hint="Test hint" validationMessage="Test validation message" validationState="success" />,
+    );
+    const input = result.getByRole('textbox');
+    const hint = result.getByText('Test hint');
+    const validationMessage = result.getByText('Test validation message');
+
+    expect(input.getAttribute('aria-describedby')).toBe(validationMessage.id + ' ' + hint.id);
+  });
+
+  it('sets aria-errormessage to the validationMessage if an error', () => {
+    const result = render(<MockField validationMessage="Test validation message" validationState="error" />);
+    const input = result.getByRole('textbox');
+    const validationMessage = result.getByText('Test validation message');
+
+    expect(validationMessage.id).toBeTruthy();
+    expect(input.getAttribute('aria-errormessage')).toBe(validationMessage.id);
+  });
+
+  it('sets aria-invalid if an error', () => {
+    const result = render(<MockField validationState="error" />);
+    const input = result.getByRole('textbox');
+
+    expect(input.getAttribute('aria-invalid')).toBeTruthy();
+  });
+
+  it('forwards ref to the primary slot', () => {
```

Comment:
Should all components forward refs to the primary slot? If so, should this be part of the conformance test?